analyses/npl/14_chirps_analysis.Rmd

```diff
@@ -18,106 +18,489 @@ library(tidyquant)
 library(extRemes)
 library(tseries)
 library(ncdf4)
+library(raster)
 
-# reading in shapefile
-nplSF <- st_read(paste0(Sys.getenv("AA_DATA_DIR"), "/private/exploration/npl/unrco/shapefiles/Major_River_Basins.shp"))
+# reading in shapefiles
+temp <- tempfile()
+unzip(zipfile = paste0(Sys.getenv("AA_DATA_DIR"), "/public/raw/npl/cod_ab/npl_cod_ab.shp.zip"), exdir = temp)
+nplSF <- st_read(temp, layer = "npl_admbnda_districts_nd_20201117")
+
+nplRBSF <- st_read(paste0(Sys.getenv("AA_DATA_DIR"), "/private/exploration/npl/unrco/shapefiles/Major_River_Basins.shp"))
+
+nplWSSF <- st_read(paste0(Sys.getenv("AA_DATA_DIR"), "/private/exploration/npl/unrco/shapefiles/Major_watershed.shp"))
 
 # reading in historical floods
 NPLHistFloods <- readxl::read_xlsx(paste0(Sys.getenv("AA_DATA_DIR"), "/private/exploration/npl/unrco/NepalHistoricalFlood1971-2020.xlsx"))
 
+# reading in GLOFAS Data
+NPLGLOFASNC <- nc_open(paste0(Sys.getenv("AA_DATA_DIR"),                        "/public/processed/npl/glofas/npl_cems-glofas-historical_v3_incorrect-coords.nc"))
+NPLGLOFAS <- tibble(Chatara = ncvar_get(NPLGLOFASNC, "Chatara_v3"),
+                    Chisapani = ncvar_get(NPLGLOFASNC, "Chisapani_v3"),
+                    Time = as.Date("1979-01-01") + (NPLGLOFASNC$dim$time$vals))
+nc_close()
+
+NPLWaterLevel <- read_csv(paste0(Sys.getenv("AA_DATA_DIR"), "/private/exploration/npl/dhm/processed/waterl_level_procssed.csv"))
+
 # function for reading in files 
 read_files <- function(filePath, fileName){
     ncObj <- nc_open(paste0(filePath, fileName))
-    DataCube <- ncvar_get(ncObj, "precipitation") %>%
+    # Karnali
+    KarnaliBox <- nplRBSF %>%
```

Comment:
Because the code for KarnaliBox and KoshiBox is the same, I would generalize it to a function where you can just pass a shapefile object, and then call it for each one.